src/Repl/Decode.tsx

```diff
@@ -0,0 +1,199 @@
+/*
+ * Decode.tsx
+ * Nov. 3rd, 2020
+ * Ben Friedman
+ *
+ * Holds decoding functionality for responses from the server
+ */
+
+
+/*
+ * decodeValue
+ * Decodes one of the possible values in a response
+ *
+ * decoded : Decoded response from the server
+ */
+const decodeValue = (decoded) => {
+  // parse by the type of the response
+  if(decoded["type"] === "Board") {
+    // Board
+    return decodeBoard(decoded["value"])
+
+  } else if(decoded["type"] === "Tuple") {
+    // Tuple
+    return decodeTuple(decoded["value"])
+
+  } else if(decoded["type"] === "Bool") {
+    // Bool
+    return decodeBool(decoded["value"])
+
+  } else if(decoded["type"] === "Int" || decoded["type"] === "Symbol") {
+    // Int || Symbol
+    return decoded["value"]
+
+  } else {
```

Comment:
I am not sure what the issue is, but there is a decoding bug. The image below is your front end changes on my back end changes:

![ben](https://user-images.githubusercontent.com/6590681/98200108-b8b0c700-1ee1-11eb-987c-1c4f4c0cfbf6.png)

I was curious to see whether my back end changes might have produced the issue, so I tried it on my front end exploratory changes. There is no issue there:

![mine](https://user-images.githubusercontent.com/6590681/98200143-cf571e00-1ee1-11eb-9c89-1ed24ad82048.png)

Here is the code:
```
game Boards

b0 : Board
b0!(x,y) = 0

b1 : Board
b1!(x,y) = 1

p : Int
p = let x = place(2, b0, (1,1)) in 
    let y = place(1, b1, (1,1)) in 1
```

Note: the reason I wrote this program was because every board that is `placed` on should be printed to the user. Good thing, because I uncovered a bug!

Reply:
Yep, that's a bug. The associated cause would be printed in the console, but that happens when the type does not match a recognized type in the decoder list. I suspect this be related to the type of Input directly? Good catch, I'll add a patch for it. That, and I don't believe all boards are printed, just the last one, but I can rectify this, and add test cases for it as well.

Reply:
Actually, looks like the output above is also wrong. I believe there should be technically double print, once for each request. First the evaluation of 'p' to bind to 'x', and secondly for doing this on the 2nd response but with the user input provided.

Reply:
New behavior with changes here. Let me know how this seems as opposed to what it did before. The double print is mentioned in the comment just above.
<img width="1415" alt="Screen Shot 2020-11-06 at 10 57 42" src="https://user-images.githubusercontent.com/5070304/98404184-fbc48480-201e-11eb-8dff-05bba01741f8.png">

Reply:
I don't think the first print should be there. It is not the value of a full REPL expression (rather, a sub-value) and it is not _after_ IO has been done with input. However, this is probably up for debate. We don't do board printing in a very principled way at this point and we can discuss over Discord.